cpp/tensorrt_llm/batch_manager/dataTransceiver.cpp

```diff
@@ -804,7 +820,15 @@ class CacheReceiver::Impl
         TLLM_CHECK_WITH_INFO(mFormatter->inquireSupport(mSelfState.getCacheState().value(), destCacheState),
             "Disagg server does not currently support these cacheState.");
 
-        RequestInfo requestInfo(requestId, mSelfState);
+        auto endpoint = contextState.getUniqueIdServerEndpoint();
+        TLLM_CHECK_WITH_INFO(endpoint.has_value(), "UniqueIdServer endpoint not found in DataTransceiverState");
+        auto const& parallelConfig = mSelfState.getCacheState().value().getParallelConfig();
+        auto const expectedRefCount = parallelConfig.mTensorParallelism * parallelConfig.mPipelineParallelism
+            * parallelConfig.mContextParallelism;
+        int32_t uniqueId = UniqueIdClient::instance().getUniqueId(
+            endpoint.value(), llmRequest.mRequestId, mServerUuid, expectedRefCount);
+
+        RequestInfo requestInfo(requestId, mSelfState, uniqueId);
```

Comment:
To support arbitrary KV cache block movements, we would need to remove the dependency on the context `requestId` no? Is the plan to make this optional?

Can we make it clear that this `requestId` is the sender `requestId` by changing the name to `senderRequestId`?

Reply:
Yes, the plan is to make this optional. I have some changes in my local branch for this which I'll create a PR for once this PR is merged.

>Can we make it clear that this requestId is the sender requestId by changing the name to senderRequestId?

Sounds good, will do.